extension/jdtls.ext/com.microsoft.gradle.bs.importer/src/com/microsoft/gradle/bs/importer/BuildTargetManager.java

```diff
@@ -0,0 +1,22 @@
+package com.microsoft.gradle.bs.importer;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.core.resources.IProject;
+
+import ch.epfl.scala.bsp4j.BuildTarget;
+
+public class BuildTargetManager {
```

Comment:
This looks like a duplicated cache with the BSP. Can we query it from BSP directly?

Reply:
Yet, another way we can do is get all build targets from build server and filter the interested one. I'll go with this approach to avoid cache as much as possible.